docs/WebTransport.md

```diff
@@ -0,0 +1,209 @@
+# Using WebTransport in Kestrel
+
+Kestrel currently implements most of the WebTransport [draft-02](https://ietf-wg-webtrans.github.io/draft-ietf-webtrans-http3/draft-ietf-webtrans-http3.html) specification, except for datagrams. Datagrams will be implemented at a later date. This document outlines how to use the functionality.
+
+# Running the sample app
+To help applications get started on implementing WebTransport, there is the `WebTransportSampleApp` project. You can find it in `src\Servers\Kestrel\samples\WebTransportSampleApp`. To use it simply run it from VS. This will launch the server and a terminal which will show logs that Kestrel prints. Now you should be able to connect to it from any client that implements the standard WebTransport draft02 specification.
+
+**Note:** Once you run the WebTransportSampleApp, it will print the certificate hash that it is using for the SSL connection. You will need to copy and paste it into your client to make sure that both the server and the client use the same one.
+
+## Using Edge or Chrome DevTools as a client
+The Chromium project has implemented a WebTransport client and can be accessed via their JS API from the Chrome or Edge DevTools console. A good sample app demoing how to use that API can be found [here](https://github.com/myjimmy/google-webtransport-sample/blob/ee13bde656c4d421d1f2a8e88fd71f572272c163/client.js).
+
+# Overview of the Kestrel WebTransport API
+## Setting up a connection
+To setup a WebTransport connection, you will first need to configure a host upon which you open a port. A very minimal example is shown below:
+```C#
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var hostBuilder = new HostBuilder()
+            .ConfigureWebHost(webHost =>
+            {
+                webHost.UseKestrel()
+                .ConfigureKestrel((context, options) =>
+                {
+                    options.Listen([SOME IP ADDRESS], [SOME PORT], listenOptions =>
+                    {
+                        listenOptions.UseHttps([YOUR CERTIFICATE]);
+                        listenOptions.Protocols = HttpProtocols.Http1AndHttp2AndHttp3;
+                    });
+                })
+                .UseStartup<Startup>();
+            });
+        var host = hostBuilder.Build();
+        host.Run();
+    }
+}
+```
+**Note:** As WebTransport uses HTTP/3, you must make sure to select the `listenOptions.UseHttps` setting as well as set the `listenOptions.UseHttps` to include HTTP/3.
```

Comment:
I think the second `UseHttps` should be `Protocols` instead.

```suggestion
**Note:** As WebTransport uses HTTP/3, you must make sure to select the `listenOptions.UseHttps` setting as well as set the `listenOptions.Protocols` to include HTTP/3.
```

Reply:
Good catch! Thanks.